metadata.md

```diff
@@ -0,0 +1,116 @@
+# Metadata Schema - Version 0.1.0
+
+In order to ensure interoperability between implementations of the Arrow in-memory format
+and associated storage formats such as Parquet and Feather, it is necessary to
+include additional top-level metadata.
+
+This information includes:
+
+-   spatial properties of each geometry column
+-   name of the primary geometry column
```

Comment:
What are the implications of the primary geometry column? R `sf` also uses a primary geometry column, but I personally think it's superfluous.

Reply:
If we have a primary geometry column, we could include a serialized spatial index in the file in the future. I think a GeoDataFrame might also have some assumptions about having a primary geometry column.

Reply:
Yes, since GeoDataFrame in geopandas has a concept of one of the geometry columns being the "active" one, we needed to serialize that state here as `primary_column`.  The primary role is to ensure that we correctly round-trip the state of the GeoDataFrame through Parquet / feather, when it has more than one geometry column.

I still need to get my head around the use cases for multiple geometry columns, but it seems theoretically appropriate to allow a spatial index to be stored per geometry column, though the most likely column would be the primary geometry.  Put differently, I'd want to make sure we're keeping the spatial index at the geometry column level rather than file level.

Reply:
I have a few example use cases for multiple geometry columns:
- Keep the data in the original format it was collected (e.g. in national grid) and have a second column in a lat-lon format.
- Keep origin and destinations in two separate columns
- Associate a geometry and a transformation, such as a point and a buffer, or a voronoi polygon
- A watershed and an outlet

For the active geometry column (maybe it could be optional), I must say it will be fully compatible with `sf` which uses the same principle and uses the first geometry column it encounters when it reads a multi-column file that has no specifications of active geometries. 

> Put differently, I'd want to make sure we're keeping the spatial index at the geometry column level rather than file level.

I agree with that principle. Aim to put everything at the column level. In my mind, this would mean moving away from `primary_column`, though. Maybe I don't understand correctly?

Reply:
`primary_column` is a hint for geopandas and other implementations that rely on the concept of a primary column as to which one was primary when the data frame was serialized.  It isn't supposed to store any additional information about that column; that should be handled via its entry in `columns`.

Is `sf` able to take a hint as to the active column at data frame construction time, or does it always default to the first one?  Either way, in the implementation in `sf` it might be a good idea to set the active column to match `primary_column` on read from Parquet / Arrow...

Reply:
Yes, `sf` can use the `primary_column` when creating the data frame.